[BUG]Wrong order of LogIt class private variables

**Describe the bug**
If rmm is used with a libary with `-Werror` then the compilation fails with the following message :
```
/home/aatish/workspace/cuhornet/hornet/../externals/rmm/include/rmm/rmm.hpp: In constructor ‘rmm::LogIt::LogIt(rmm::Logger::MemEvent_t, void*, size_t, cudaStream_t, const char*, unsigned int, bool)’:
/home/aatish/workspace/cuhornet/hornet/../externals/rmm/include/rmm/rmm.hpp:101:8: error: ‘rmm::LogIt::usageLogging’ will be initialized after [-Werror=reorder]
   bool usageLogging;
        ^~~~~~~~~~~~
/home/aatish/workspace/cuhornet/hornet/../externals/rmm/include/rmm/rmm.hpp:100:16: error:   ‘unsigned int rmm::LogIt::line’ [-Werror=reorder]
   unsigned int line;
                ^~~~
/home/aatish/workspace/cuhornet/hornet/../externals/rmm/include/rmm/rmm.hpp:59:3: error:   when initialized here [-Werror=reorder]
   LogIt(Logger::MemEvent_t event, void* ptr, size_t size, cudaStream_t stream,
   ^~~~~
cc1plus: all warnings being treated as errors
```
This can be replicated with branch-0.10

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 
 ## Bug Fixes
 
+ - PR #125 Fixed order of private variables in LogIt
 
 # RMM 0.9.0 (Date TBD)
 

--- a/include/rmm/rmm.hpp
+++ b/include/rmm/rmm.hpp
@@ -64,8 +64,8 @@ class LogIt {
         ptr(ptr),
         size(size),
         stream(stream),
-        usageLogging(usageLogging),
-        line(line) {
+        line(line),
+        usageLogging(usageLogging) {
     if (filename) file = filename;
     if (Manager::getOptions().enable_logging) {
       cudaGetDevice(&device);
@@ -97,8 +97,8 @@ class LogIt {
   cudaStream_t stream;
   rmm::Logger::TimePt start;
   std::string file;
-  bool usageLogging;
   unsigned int line;
+  bool usageLogging;
 };
 
 /** ---------------------------------------------------------------------------*

--- a/include/rmm/rmm.hpp
+++ b/include/rmm/rmm.hpp
@@ -97,8 +97,8 @@ class LogIt {
   cudaStream_t stream;
   rmm::Logger::TimePt start;
   std::string file;
-  unsigned int line;
   bool usageLogging;
+  unsigned int line;
 };
 
 /** ---------------------------------------------------------------------------*

